hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -216,11 +220,29 @@ public Result inputSplits(
     final String endInstant = instantToIssue.getTimestamp();
     final AtomicInteger cnt = new AtomicInteger(0);
     final String mergeType = this.conf.getString(FlinkOptions.MERGE_TYPE);
+    final FileSystem fs = FSUtils.getFs(path.toString(), hadoopConf);
     List<MergeOnReadInputSplit> inputSplits = writePartitions.stream()
         .map(relPartitionPath -> fsView.getLatestMergedFileSlicesBeforeOrOn(relPartitionPath, endInstant)
+            .filter(fileSlice -> {
+              Option<String> basePath = fileSlice.getBaseFile().map(BaseFile::getPath);
+              try {
+                return basePath.isPresent() && fs.exists(new org.apache.hadoop.fs.Path(basePath.get()));
+              } catch (IOException e) {
+                LOG.error("Checking exists of base path: {} error", basePath);
```

Comment:
Can we write a tool method for `FileSlice` by checking file existence by filtering out the non-exist files together ? Maybe name it `filterFileSliceWithValidFiles`

Reply:
Yes, it's a better way.